pycloudlib/ec2/cloud.py

```diff
@@ -242,9 +242,9 @@ def snapshot(self, instance, clean=True):
         image_ami_edited = response['ImageId']
         image = self.resource.Image(image_ami_edited)
 
-        self._wait_for_snapshot(image)
-
-        instance.start(wait=True)
+        if wait:
+            self._wait_for_snapshot(image)
+            instance.start(wait=True)
```

Comment:
I think we need to pull this back out of if scope, otherwise we don't restart this instance in non-wait cases
```suggestion
        instance.start(wait=wait)
```

Reply:
We're stopping the image to make a snapshot, so I'm assuming we can't start the instance while the snapshot is still being created. I figured if you didn't want to wait you'd just restart the instance manually, but now that I'm thinking about it again, that's not going to be super obvious. We could spawn a background thread to start it once the snapshot is created, but not sure that's a great idea anyway. There's also the option of status callbacks...

Maybe snapshot shouldn't support a wait argument? Or maybe we keep the argument in, but for now only provide the waiting code and raise a `NotImplementedError` if `wait=False` until we decide to get around to it?

Reply:
@TheRealFalcon I see you implemented that logic in OCI, harm I'd lean toward either raising NoImplementedError on wait=False for all other pycloud subclasses or not supporting wait param on any and just waiting if you are creating an image from a running instance. In most cloud cases, the source instance from which we are creating a snapshot will be brought down anyway, and bringing it up mid-image creation is likely to cause issues anyway I'd guess.